checker/tests/lock/FlowExpressionsTest.java

```diff
@@ -18,7 +18,6 @@
   public void method() {
     // :: error: (lock.not.held)
     getm().field = new Object();
-    // :: error: (lock.not.held)
```

Comment:
I think this warning should be issued. The type of `m` is `@GuardedBy("<self>")` and the lock `m` isn't held.  I think the problem is that `m` is being refined it `@NewObject` because it's a final field.